Installation issue: Dealii

### Steps to reproduce the issue

```console
$ spack env create dealii
$ spack env activate -p dealii
[dealii]$ spack spec -l dealii
Input spec
--------------------------------
dealii

Concretized
--------------------------------
qyeizap  dealii@9.4.0%gcc@12.1.1+adol-c+arborx+arpack+assimp+cgal~cuda~doc+examples+ginkgo+gmsh+gsl+hdf5~int64~ipo+metis+mpi+muparser~nanoflann~netcdf+oce~optflags+p4est+petsc~python+scalapack+simplex+slepc+sundials+symengine+threads+trilinos build_type=DebugRelease cxxstd=default arch=linux-fedora36-skylake
r6lsjzh      ^adol-c@2.7.2%gcc@12.1.1~advanced_branching+atrig_erf~boost+doc+examples~openmp~sparse+stdczero+traceless_refcounting patches=715b878 arch=linux-fedora36-skylake
7feq6qw      ^arborx@1.1%gcc@12.1.1~cuda~ipo+mpi~openmp~rocm+serial~sycl+trilinos build_type=RelWithDebInfo arch=linux-fedora36-skylake
uv2c6xy          ^cmake@3.23.2%gcc@12.1.1~doc+ncurses+ownlibs~qt build_type=Release arch=linux-fedora36-skylake
xig7hbj              ^ncurses@6.2%gcc@12.1.1~symlinks+termlib abi=none arch=linux-fedora36-skylake
5elkw2l                  ^pkgconf@1.8.0%gcc@12.1.1 arch=linux-fedora36-skylake
b6vbqg6              ^openssl@1.1.1q%gcc@12.1.1~docs~shared certs=mozilla patches=3fdcf2d arch=linux-fedora36-skylake
wuzfc5h                  ^ca-certificates-mozilla@2022-07-19%gcc@12.1.1 arch=linux-fedora36-skylake
qbc7rdo                  ^perl@5.34.1%gcc@12.1.1+cpanm+shared+threads arch=linux-fedora36-skylake
c5r42si                      ^berkeley-db@18.1.40%gcc@12.1.1+cxx~docs+stl patches=b231fcc arch=linux-fedora36-skylake
l345lio                      ^bzip2@1.0.8%gcc@12.1.1~debug~pic+shared arch=linux-fedora36-skylake
u7rjr7v                          ^diffutils@3.8%gcc@12.1.1 arch=linux-fedora36-skylake
nqoshzw                              ^libiconv@1.16%gcc@12.1.1 libs=shared,static arch=linux-fedora36-skylake
mlmz5u5                      ^gdbm@1.19%gcc@12.1.1 arch=linux-fedora36-skylake
sqsnmam                          ^readline@8.1.2%gcc@12.1.1 arch=linux-fedora36-skylake
udnwhqr                      ^zlib@1.2.12%gcc@12.1.1+optimize+pic+shared patches=0d38234 arch=linux-fedora36-skylake
npqerut          ^openmpi@4.1.4%gcc@12.1.1~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~java~legacylaunchers~lustre~memchecker+romio+rsh~singularity+static+vt+wrapper-rpath fabrics=none schedulers=none arch=linux-fedora36-skylake
fy3v7ds              ^hwloc@2.7.1%gcc@12.1.1~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm+shared arch=linux-fedora36-skylake
ih62fj5                  ^libpciaccess@0.16%gcc@12.1.1 arch=linux-fedora36-skylake
3567hmt                      ^libtool@2.4.7%gcc@12.1.1 arch=linux-fedora36-skylake
jddjki2                          ^m4@1.4.19%gcc@12.1.1+sigsegv patches=9dc5fbd,bfdffa7 arch=linux-fedora36-skylake
45sbzou                              ^libsigsegv@2.13%gcc@12.1.1 arch=linux-fedora36-skylake
ed2soq3                      ^util-macros@1.19.3%gcc@12.1.1 arch=linux-fedora36-skylake
wtr45c7                  ^libxml2@2.9.13%gcc@12.1.1~python arch=linux-fedora36-skylake
japw4fd                      ^xz@5.2.5%gcc@12.1.1~pic libs=shared,static arch=linux-fedora36-skylake
yaav3oj              ^numactl@2.0.14%gcc@12.1.1 patches=4e1d78c,62fc8a8,ff37630 arch=linux-fedora36-skylake
ca3mybb                  ^autoconf@2.69%gcc@12.1.1 patches=35c4492,7793209,a49dd5b arch=linux-fedora36-skylake
ef7cdl4                  ^automake@1.16.5%gcc@12.1.1 arch=linux-fedora36-skylake
b3fgazq              ^openssh@9.0p1%gcc@12.1.1+gssapi arch=linux-fedora36-skylake
7nenwx7                  ^krb5@1.19.3%gcc@12.1.1+shared arch=linux-fedora36-skylake
ew35doc                      ^bison@3.8.2%gcc@12.1.1 arch=linux-fedora36-skylake
7azf4j7                      ^gettext@0.21%gcc@12.1.1+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-fedora36-skylake
k7slh3q                          ^tar@1.34%gcc@12.1.1 zip=pigz arch=linux-fedora36-skylake
stwhfns                              ^pigz@2.7%gcc@12.1.1 arch=linux-fedora36-skylake
vaae2fh                              ^zstd@1.5.2%gcc@12.1.1+programs compression=none libs=shared,static arch=linux-fedora36-skylake
m4zm5pk                  ^libedit@3.1-20210216%gcc@12.1.1 arch=linux-fedora36-skylake
k6yjho5              ^pmix@4.1.2%gcc@12.1.1~docs+pmi_backwards_compatibility~restful arch=linux-fedora36-skylake
7n3n5io                  ^libevent@2.1.12%gcc@12.1.1+openssl arch=linux-fedora36-skylake
brjcnu4          ^trilinos@13.0.1%gcc@12.1.1~adelus~adios2+amesos+amesos2+anasazi+aztec~basker+belos~boost~chaco~complex~cuda~cuda_rdc~debug~dtk+epetra+epetraext~epetraextbtf~epetraextexperimental~epetraextgraphreorderings~exodus+explicit_template_instantiation~float+fortran~gtest~hdf5~hypre+ifpack+ifpack2~intrepid~intrepid2~ipo~isorropia+kokkos~mesquite~minitensor+ml+mpi+muelu~mumps~nox~openmp~panzer~phalanx~piro~python~rocm~rocm_rdc~rol~rythmos+sacado~scorec~shards+shared~shylu~stk~stokhos~stratimikos~strumpack~suite-sparse~superlu~superlu-dist~teko~tempus~thyra+tpetra~trilinoscouplings~wrapper~x11~zoltan~zoltan2 build_type=RelWithDebInfo cxxstd=14 gotype=long_long arch=linux-fedora36-skylake
74q4dkd              ^openblas@0.3.20%gcc@12.1.1~bignuma~consistent_fpcsr~ilp64+locking+pic+shared symbol_suffix=none threads=none arch=linux-fedora36-skylake
2dx2qxy      ^arpack-ng@3.8.0%gcc@12.1.1+mpi+shared arch=linux-fedora36-skylake
glz6odw      ^assimp@5.2.4%gcc@12.1.1~ipo+shared build_type=RelWithDebInfo arch=linux-fedora36-skylake
j6gvcm6      ^boost@1.79.0%gcc@12.1.1+atomic+chrono~clanglibcpp+container~context~contract~coroutine+date_time~debug+exception~fiber+filesystem+graph~graph_parallel~icu+iostreams~json+locale+log+math~mpi+multithreaded~nowide~numpy~pic+program_options~python+random+regex+serialization+shared+signals~singlethreaded~stacktrace+system~taggedlayout+test+thread+timer~type_erasure~versionedlayout+wave cxxstd=98 patches=a440f96 visibility=hidden arch=linux-fedora36-skylake
bisxhj7      ^cgal@5.4.1%gcc@12.1.1~core~demos+eigen~header_only~imageio~ipo+shared build_type=Release arch=linux-fedora36-skylake
bkjtwmm          ^eigen@3.4.0%gcc@12.1.1~ipo build_type=RelWithDebInfo arch=linux-fedora36-skylake
7bbyjzb          ^gmp@6.2.1%gcc@12.1.1 libs=shared,static arch=linux-fedora36-skylake
4kz7feu          ^mpfr@4.1.0%gcc@12.1.1 libs=shared,static arch=linux-fedora36-skylake
slotcdl              ^autoconf-archive@2022.02.11%gcc@12.1.1 patches=130cd48 arch=linux-fedora36-skylake
ovgls2l              ^texinfo@6.5%gcc@12.1.1 patches=12f6edb,1732115 arch=linux-fedora36-skylake
huyjxd2      ^ginkgo@1.4.0%gcc@12.1.1~cuda~develtools~full_optimizations~hwloc~ipo~oneapi+openmp~rocm+shared build_type=Release patches=43e571a arch=linux-fedora36-skylake
6vc4uuu      ^gmsh@4.8.4%gcc@12.1.1+alglib~cairo+cgns+compression~eigen~external+fltk+gmp~hdf5~ipo+med+metis+mmg~mpi+netgen+oce~opencascade~openmp~petsc~privateapi+shared~slepc+tetgen+voropp build_type=RelWithDebInfo arch=linux-fedora36-skylake
3bghe5v          ^cgns@4.3.0%gcc@12.1.1~base_scope~fortran+hdf5~int64~ipo~legacy~mem_debug+mpi+scoping+shared~static~testing build_type=RelWithDebInfo arch=linux-fedora36-skylake
li5cjeh              ^hdf5@1.10.7%gcc@12.1.1~cxx+fortran+hl~ipo~java+mpi+shared~szip~threadsafe+tools api=default build_type=RelWithDebInfo patches=2a1e311 arch=linux-fedora36-skylake
coju4le          ^fltk@1.3.7%gcc@12.1.1+gl+shared~xft arch=linux-fedora36-skylake
wnzf7wb              ^glx@1.4%gcc@12.1.1 arch=linux-fedora36-skylake
bkp7mwy                  ^mesa@22.1.2%gcc@12.1.1+glx+llvm+opengl~opengles+osmesa~strip buildtype=release default_library=shared patches=ada85be arch=linux-fedora36-skylake
ud3td3e                      ^expat@2.4.8%gcc@12.1.1+libbsd arch=linux-fedora36-skylake
2wigrz5                          ^libbsd@0.11.5%gcc@12.1.1 arch=linux-fedora36-skylake
vli3emk                              ^libmd@1.0.4%gcc@12.1.1 arch=linux-fedora36-skylake
ftyxnj5                      ^flex@2.6.4%gcc@12.1.1+lex~nls patches=f8b85a0 arch=linux-fedora36-skylake
h775ioq                          ^findutils@4.9.0%gcc@12.1.1 patches=440b954 arch=linux-fedora36-skylake
rxumd2g                          ^help2man@1.47.16%gcc@12.1.1 arch=linux-fedora36-skylake
aw7okev                      ^glproto@1.4.17%gcc@12.1.1 arch=linux-fedora36-skylake
h5hmswj                      ^libunwind@1.6.2%gcc@12.1.1~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib components=none libs=shared,static arch=linux-fedora36-skylake
ij2hqg4                      ^libx11@1.7.0%gcc@12.1.1 arch=linux-fedora36-skylake
pxr2m5y                          ^inputproto@2.3.2%gcc@12.1.1 arch=linux-fedora36-skylake
ccbqj5g                          ^kbproto@1.0.7%gcc@12.1.1 arch=linux-fedora36-skylake
7hheqsj                          ^libxcb@1.14%gcc@12.1.1 arch=linux-fedora36-skylake
7rd5rh6                              ^libpthread-stubs@0.4%gcc@12.1.1 arch=linux-fedora36-skylake
2k2yk5b                              ^libxau@1.0.8%gcc@12.1.1 arch=linux-fedora36-skylake
g5t442g                                  ^xproto@7.0.31%gcc@12.1.1 arch=linux-fedora36-skylake
v3qlmwi                              ^libxdmcp@1.1.2%gcc@12.1.1 arch=linux-fedora36-skylake
hn6xk3s                              ^xcb-proto@1.14.1%gcc@12.1.1 arch=linux-fedora36-skylake
2af4t4x                          ^xextproto@7.3.0%gcc@12.1.1 arch=linux-fedora36-skylake
r5tzcnn                          ^xtrans@1.3.5%gcc@12.1.1 arch=linux-fedora36-skylake
uwzqs25                      ^libxext@1.3.3%gcc@12.1.1 arch=linux-fedora36-skylake
bhmoi7d                      ^libxt@1.1.5%gcc@12.1.1 arch=linux-fedora36-skylake
nggoe3v                          ^libice@1.0.9%gcc@12.1.1 arch=linux-fedora36-skylake
63a3per                          ^libsm@1.2.3%gcc@12.1.1 arch=linux-fedora36-skylake
tqzhmqp                              ^util-linux-uuid@2.37.4%gcc@12.1.1 arch=linux-fedora36-skylake
udamuws                      ^llvm@14.0.6%gcc@12.1.1+clang+compiler-rt~cuda~flang+gold+internal_unwind~ipo+libcxx~link_llvm_dylib+lld+lldb+llvm_dylib~mlir+omp_as_runtime~omp_debug~omp_tsan+polly~python~split_dwarf~z3 build_type=Release patches=d85ef51,f920173 shlib_symbol_version=none targets=none version_suffix=none arch=linux-fedora36-skylake
2cnvzlk                          ^binutils@2.38%gcc@12.1.1~gas+gold~headers~interwork+ld~libiberty~lto+nls+plugins libs=shared,static arch=linux-fedora36-skylake
nmkgojt                          ^ninja@1.11.0%gcc@12.1.1 arch=linux-fedora36-skylake
cvjsqj5                              ^python@3.9.13%gcc@12.1.1+bz2+ctypes+dbm~debug+ensurepip+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,f2fd060 arch=linux-fedora36-skylake
ircarqj                                  ^libffi@3.4.2%gcc@12.1.1 arch=linux-fedora36-skylake
d7k2wyt                                  ^sqlite@3.38.5%gcc@12.1.1+column_metadata+dynamic_extensions+fts~functions+rtree arch=linux-fedora36-skylake
zneldo5                          ^perl-data-dumper@2.173%gcc@12.1.1 arch=linux-fedora36-skylake
6r5oxw7                      ^meson@0.62.2%gcc@12.1.1 patches=9c87472,aa6c50d arch=linux-fedora36-skylake
w2ju2ze                          ^py-pip@22.1.2%gcc@12.1.1 arch=linux-fedora36-skylake
ytfqoiu                          ^py-setuptools@63.0.0%gcc@12.1.1 arch=linux-fedora36-skylake
og67ohe                          ^py-wheel@0.37.1%gcc@12.1.1 arch=linux-fedora36-skylake
y7o2tc7                      ^py-mako@1.1.6%gcc@12.1.1 arch=linux-fedora36-skylake
lh3dtan                          ^py-markupsafe@2.0.1%gcc@12.1.1 arch=linux-fedora36-skylake
sdgulol                      ^xrandr@1.5.0%gcc@12.1.1 arch=linux-fedora36-skylake
r4xjxm7                          ^libxrandr@1.5.0%gcc@12.1.1 arch=linux-fedora36-skylake
wgx3aqm                              ^libxrender@0.9.10%gcc@12.1.1 arch=linux-fedora36-skylake
vtpqgng                                  ^renderproto@0.11.1%gcc@12.1.1 arch=linux-fedora36-skylake
fogyleh                              ^randrproto@1.5.0%gcc@12.1.1 arch=linux-fedora36-skylake
dfnxoba          ^freetype@2.11.1%gcc@12.1.1 arch=linux-fedora36-skylake
46556it              ^libpng@1.6.37%gcc@12.1.1 arch=linux-fedora36-skylake
zyun63j          ^med@4.0.0%gcc@12.1.1+api23~fortran~ipo+mpi~shared build_type=RelWithDebInfo patches=ba35197 arch=linux-fedora36-skylake
op57pjr          ^mesa-glu@9.0.2%gcc@12.1.1 gl=glx arch=linux-fedora36-skylake
hdctwpz          ^mmg@5.6.0%gcc@12.1.1~doc~ipo+scotch+shared~vtk build_type=RelWithDebInfo arch=linux-fedora36-skylake
v7slohx              ^scotch@7.0.1%gcc@12.1.1+compression~esmumps~int64~ipo~metis+mpi+shared build_type=RelWithDebInfo arch=linux-fedora36-skylake
65ain5y          ^oce@0.18.3%gcc@12.1.1~X11+tbb arch=linux-fedora36-skylake
kjtiuyu              ^intel-tbb@2020.3%gcc@12.1.1~ipo+shared+tm build_type=RelWithDebInfo cxxstd=default patches=62ba015,ce1fb16,d62cb66 arch=linux-fedora36-skylake
hojxhn6      ^gsl@2.7.1%gcc@12.1.1~external-cblas arch=linux-fedora36-skylake
k5ine2z      ^metis@5.1.0%gcc@12.1.1~gdb~int64~real64+shared build_type=Release patches=4991da9,b1225da arch=linux-fedora36-skylake
6af2pf7      ^muparser@2.2.6.1%gcc@12.1.1 arch=linux-fedora36-skylake
subtwjw      ^netlib-scalapack@2.2.0%gcc@12.1.1~ipo~pic+shared build_type=Release patches=072b006,1c9ce5f arch=linux-fedora36-skylake
lrocyfn      ^p4est@2.8%gcc@12.1.1+mpi~openmp arch=linux-fedora36-skylake
cdtf5xy      ^petsc@3.17.3%gcc@12.1.1~X~batch~cgns~complex~cuda~debug+double~exodusii~fftw+fortran~giflib+hdf5~hpddm~hwloc+hypre~int64~jpeg~knl~kokkos~libpng~libyaml~memkind+metis~mkl-pardiso~mmg~moab~mpfr+mpi~mumps~openmp~p4est~parmmg~ptscotch~random123~rocm~saws~scalapack+shared~strumpack~suite-sparse+superlu-dist~tetgen~trilinos~valgrind clanguage=C arch=linux-fedora36-skylake
prt6snl          ^hypre@2.25.0%gcc@12.1.1~complex~cuda~debug+fortran~gptune~int64~internal-superlu~mixedint+mpi~openmp~rocm+shared~superlu-dist~unified-memory arch=linux-fedora36-skylake
fsslt63          ^parmetis@4.0.3%gcc@12.1.1~gdb~int64~ipo+shared build_type=RelWithDebInfo patches=4f89253,50ed208,704b84f arch=linux-fedora36-skylake
nge72oq          ^superlu-dist@7.2.0%gcc@12.1.1~cuda~int64~ipo~openmp~rocm+shared build_type=RelWithDebInfo patches=8da9e21 arch=linux-fedora36-skylake
gvmfmws      ^slepc@3.17.1%gcc@12.1.1+arpack~blopex~cuda~rocm arch=linux-fedora36-skylake
x3o6nxm      ^suite-sparse@5.10.1%gcc@12.1.1~cuda~graphblas~openmp+pic~tbb arch=linux-fedora36-skylake
s3jnhra      ^sundials@6.2.0%gcc@12.1.1+ARKODE+CVODE+CVODES+IDA+IDAS+KINSOL~cuda+examples+examples-install~f2003~fcmix+generic-math~hypre~int64~ipo~klu~lapack~magma~monitoring+mpi~openmp~petsc~profiling~pthread~raja~rocm+shared+static~superlu-dist~superlu-mt~sycl~trilinos build_type=RelWithDebInfo cstd=99 cxxstd=14 logging-level=0 precision=double arch=linux-fedora36-skylake
vlgyeov      ^symengine@0.9.0%gcc@12.1.1~boostmp~flint~ipo~llvm+mpc+mpfr~openmp~piranha+shared+thread_safe build_type=Release arch=linux-fedora36-skylake
j4rzqyn          ^cereal@1.3.2%gcc@12.1.1~ipo build_type=RelWithDebInfo patches=2dfa0bf arch=linux-fedora36-skylake
cwmmhkt          ^mpc@1.2.1%gcc@12.1.1 libs=shared,static arch=linux-fedora36-skylake

[dealii]$ spack install dealii
```

### Error message

<details><summary>Error message</summary><pre>
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Ninja' '-DCMAKE_INSTALL_PREFIX:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/dealii-9.4.0-qyeizaptfk4artq6dukrk5scxsbdlbsc' '-DCMAKE_BUILD_TYPE:STRING=DebugRelease' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/dealii-9.4.0-qyeizaptfk4artq6dukrk5scxsbdlbsc/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/dealii-9.4.0-qyeizaptfk4artq6dukrk5scxsbdlbsc/lib64;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/arborx-1.1-7feq6qw55a5vz7pbj7chszo6wednowk2/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/boost-1.79.0-j6gvcm6giqbv5v35zp5mxov5z7j6lzxk/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/gsl-2.7.1-hojxhn65gv6apjansn3n3omymiafsh73/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/hdf5-1.10.7-li5cjehgcpyndpayoizmkefgixiflylp/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/intel-tbb-2020.3-kjtiuyurwoknk4qw5sx3ad2j3omtzfs5/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/metis-5.1.0-k5ine2zjb726cczsbtz6pgphd5yjdpul/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/netlib-scalapack-2.2.0-subtwjwg2ncyraoc7vesgzymzkllzdrd/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/oce-0.18.3-65ain5ym3j33g4wtjrx7ftntd3fkin26/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openblas-0.3.20-74q4dkdrckytcm5hzrnnrfutkpkerlyu/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openmpi-4.1.4-npqeruttdpenhdgkcqjm5lsvv6b5xacb/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/p4est-2.8-lrocyfndffdztw6hnjwwmpdjdhcv4w45/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/petsc-3.17.3-cdtf5xyb6g565apdyeisl232fmhjs4q2/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/slepc-3.17.1-gvmfmwsa4kndqv3grga7qh547jw44cg7/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/suite-sparse-5.10.1-x3o6nxm7l7tfylijw4tipj2ag57md6u6/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/symengine-0.9.0-vlgyeovgk5dxqf4ubhndclgfovshswl2/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/trilinos-13.0.1-brjcnu4dd6aj442mnlmjmjdfnidaraam/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/zlib-1.2.12-udnwhqrtgcagshbyic77wjkfkddmkrd2/lib;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/adol-c-2.7.2-r6lsjzhut35x44zvc4swz66ug6othhu3/lib64;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/arpack-ng-3.8.0-2dx2qxyg7mj3bhg57eageyoaevb6edzv/lib64;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/assimp-5.2.4-glz6odwrgj3lmezurno5p676hbq6d5qa/lib64;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/cgal-5.4.1-bisxhj7znc4xqzaadnfuhgdpc4kayyal/lib64;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/ginkgo-1.4.0-huyjxd26eqlb5lbr25mgcxspkbtff6r2/lib64;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/muparser-2.2.6.1-6af2pf737zebqcpe47zaqxr3uytgkjgd/lib64;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/sundials-6.2.0-s3jnhra6j6riy6hxaomhhxd35po3a43u/lib64;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/symengine-0.9.0-vlgyeovgk5dxqf4ubhndclgfovshswl2/lib64' '-DCMAKE_PREFIX_PATH:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/symengine-0.9.0-vlgyeovgk5dxqf4ubhndclgfovshswl2;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/mpc-1.2.1-cwmmhktylhmibmh6ji444yp7ere5pfuy;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/cereal-1.3.2-j4rzqynsuvxtjglovblql22lt4zl3hnu;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/sundials-6.2.0-s3jnhra6j6riy6hxaomhhxd35po3a43u;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/suite-sparse-5.10.1-x3o6nxm7l7tfylijw4tipj2ag57md6u6;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/slepc-3.17.1-gvmfmwsa4kndqv3grga7qh547jw44cg7;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/petsc-3.17.3-cdtf5xyb6g565apdyeisl232fmhjs4q2;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/superlu-dist-7.2.0-nge72oqy7unvqrdg24hgm7peemi56onw;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/parmetis-4.0.3-fsslt63nlpqgw6ratq3rfyjy3w5yhnh7;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/hypre-2.25.0-prt6snlkg5wzyexhkqjzumojhtvkh47b;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/p4est-2.8-lrocyfndffdztw6hnjwwmpdjdhcv4w45;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/netlib-scalapack-2.2.0-subtwjwg2ncyraoc7vesgzymzkllzdrd;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/muparser-2.2.6.1-6af2pf737zebqcpe47zaqxr3uytgkjgd;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/metis-5.1.0-k5ine2zjb726cczsbtz6pgphd5yjdpul;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/gsl-2.7.1-hojxhn65gv6apjansn3n3omymiafsh73;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/gmsh-4.8.4-6vc4uuuxqlpaymtde2acgofq7mimz7qk;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/oce-0.18.3-65ain5ym3j33g4wtjrx7ftntd3fkin26;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/intel-tbb-2020.3-kjtiuyurwoknk4qw5sx3ad2j3omtzfs5;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/ninja-1.11.0-nmkgojteo5zgin7x7jw2mxizkhsbtrdc;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/hdf5-1.10.7-li5cjehgcpyndpayoizmkefgixiflylp;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/ginkgo-1.4.0-huyjxd26eqlb5lbr25mgcxspkbtff6r2;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/cgal-5.4.1-bisxhj7znc4xqzaadnfuhgdpc4kayyal;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/mpfr-4.1.0-4kz7feuwlhl47o34tkaoi5fgabth6mny;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/gmp-6.2.1-7bbyjzbmmfsnxlbfzlaaozf6rpc64yau;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/eigen-3.4.0-bkjtwmm6o3jkv24ozwskvmvphyhz4wym;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/boost-1.79.0-j6gvcm6giqbv5v35zp5mxov5z7j6lzxk;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/assimp-5.2.4-glz6odwrgj3lmezurno5p676hbq6d5qa;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/arpack-ng-3.8.0-2dx2qxyg7mj3bhg57eageyoaevb6edzv;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/arborx-1.1-7feq6qw55a5vz7pbj7chszo6wednowk2;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/trilinos-13.0.1-brjcnu4dd6aj442mnlmjmjdfnidaraam;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openblas-0.3.20-74q4dkdrckytcm5hzrnnrfutkpkerlyu;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openmpi-4.1.4-npqeruttdpenhdgkcqjm5lsvv6b5xacb;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/pmix-4.1.2-k6yjho5lyrhi5ugxkawgjghnxbyqx6me;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/libevent-2.1.12-7n3n5ior5tjp5mzlkawqjytyysbidwll;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/numactl-2.0.14-yaav3ojfnjulxojgc3gedqct5kvq33ob;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/hwloc-2.7.1-fy3v7dssfwcq7xhiuthdnhdqnijic5hd;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/libxml2-2.9.13-wtr45c7e6qa6vsdn3dl7apwow367r5hl;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/xz-5.2.5-japw4fde5rmdfyftmvajdngxythw3syx;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/libpciaccess-0.16-ih62fj5dtcdj32rzjjec3ptpet5rzsi6;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/cmake-3.23.2-uv2c6xywg4p2qbki7ivpyl65gsbda6ab;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openssl-1.1.1q-b6vbqg64ubk3yxxm5lqxtg5g2krieqtr;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/zlib-1.2.12-udnwhqrtgcagshbyic77wjkfkddmkrd2;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/bzip2-1.0.8-l345lio2jvjnlz3rwqhorqbfqldqvw6b;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/libiconv-1.16-nqoshzwcdvdq5l5md645ncb4d7artg4n;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/ncurses-6.2-xig7hbjtvrsnayigz32ok4lqbkcgyvsh;/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/adol-c-2.7.2-r6lsjzhut35x44zvc4swz66ug6othhu3' '-DDEAL_II_EXAMPLES_RELDIR:STRING=share/deal.II/examples' '-DDEAL_II_DOCREADME_RELDIR:STRING=share/deal.II/' '-DDEAL_II_DOCHTML_RELDIR:STRING=share/deal.II/doc' '-DBOOST_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/boost-1.79.0-j6gvcm6giqbv5v35zp5mxov5z7j6lzxk' '-DLAPACK_FOUND:BOOL=ON' '-DLAPACK_INCLUDE_DIRS:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openblas-0.3.20-74q4dkdrckytcm5hzrnnrfutkpkerlyu/include' '-DLAPACK_LIBRARIES:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openblas-0.3.20-74q4dkdrckytcm5hzrnnrfutkpkerlyu/lib/libopenblas.so' '-DUMFPACK_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/suite-sparse-5.10.1-x3o6nxm7l7tfylijw4tipj2ag57md6u6' '-DZLIB_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/zlib-1.2.12-udnwhqrtgcagshbyic77wjkfkddmkrd2' '-DDEAL_II_ALLOW_BUNDLED:BOOL=OFF' '-DDEAL_II_COMPONENT_DOCUMENTATION:BOOL=OFF' '-DDEAL_II_COMPONENT_EXAMPLES:BOOL=ON' '-DDEAL_II_WITH_64BIT_INDICES:BOOL=OFF' '-DDEAL_II_WITH_CUDA:BOOL=OFF' '-DDEAL_II_WITH_MPI:BOOL=ON' '-DCMAKE_C_COMPILER:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openmpi-4.1.4-npqeruttdpenhdgkcqjm5lsvv6b5xacb/bin/mpicc' '-DCMAKE_CXX_COMPILER:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openmpi-4.1.4-npqeruttdpenhdgkcqjm5lsvv6b5xacb/bin/mpic++' '-DCMAKE_Fortran_COMPILER:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openmpi-4.1.4-npqeruttdpenhdgkcqjm5lsvv6b5xacb/bin/mpif90' '-DMPI_C_COMPILER:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openmpi-4.1.4-npqeruttdpenhdgkcqjm5lsvv6b5xacb/bin/mpicc' '-DMPI_CXX_COMPILER:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openmpi-4.1.4-npqeruttdpenhdgkcqjm5lsvv6b5xacb/bin/mpic++' '-DMPI_Fortran_COMPILER:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/openmpi-4.1.4-npqeruttdpenhdgkcqjm5lsvv6b5xacb/bin/mpif90' '-DMPI_CXX_LINK_FLAGS:STRING=-lopen-pal' '-DDEAL_II_COMPONENT_PYTHON_BINDINGS:BOOL=OFF' '-DDEAL_II_WITH_SIMPLEX_SUPPORT:BOOL=ON' '-DDEAL_II_WITH_TBB:BOOL=ON' '-DTBB_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/intel-tbb-2020.3-kjtiuyurwoknk4qw5sx3ad2j3omtzfs5' '-DDEAL_II_WITH_GSL:BOOL=ON' '-DGSL_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/gsl-2.7.1-hojxhn65gv6apjansn3n3omymiafsh73' '-DDEAL_II_WITH_HDF5:BOOL=ON' '-DHDF5_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/hdf5-1.10.7-li5cjehgcpyndpayoizmkefgixiflylp' '-DDEAL_II_WITH_P4EST:BOOL=ON' '-DP4EST_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/p4est-2.8-lrocyfndffdztw6hnjwwmpdjdhcv4w45' '-DDEAL_II_WITH_PETSC:BOOL=ON' '-DPETSC_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/petsc-3.17.3-cdtf5xyb6g565apdyeisl232fmhjs4q2' '-DDEAL_II_WITH_SLEPC:BOOL=ON' '-DSLEPC_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/slepc-3.17.1-gvmfmwsa4kndqv3grga7qh547jw44cg7' '-DDEAL_II_WITH_TRILINOS:BOOL=ON' '-DTRILINOS_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/trilinos-13.0.1-brjcnu4dd6aj442mnlmjmjdfnidaraam' '-DDEAL_II_WITH_METIS:BOOL=ON' '-DMETIS_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/metis-5.1.0-k5ine2zjb726cczsbtz6pgphd5yjdpul' '-DDEAL_II_WITH_SUNDIALS:BOOL=ON' '-DSUNDIALS_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/sundials-6.2.0-s3jnhra6j6riy6hxaomhhxd35po3a43u' '-DDEAL_II_WITH_NANOFLANN:BOOL=OFF' '-DDEAL_II_WITH_ASSIMP:BOOL=ON' '-DASSIMP_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/assimp-5.2.4-glz6odwrgj3lmezurno5p676hbq6d5qa' '-DDEAL_II_WITH_GMSH:BOOL=ON' '-DGMSH_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/gmsh-4.8.4-6vc4uuuxqlpaymtde2acgofq7mimz7qk' '-DDEAL_II_WITH_MUPARSER:BOOL=ON' '-DMUPARSER_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/muparser-2.2.6.1-6af2pf737zebqcpe47zaqxr3uytgkjgd' '-DDEAL_II_WITH_SYMENGINE:BOOL=ON' '-DSYMENGINE_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/symengine-0.9.0-vlgyeovgk5dxqf4ubhndclgfovshswl2' '-DDEAL_II_WITH_GINKGO:BOOL=ON' '-DGINKGO_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/ginkgo-1.4.0-huyjxd26eqlb5lbr25mgcxspkbtff6r2' '-DDEAL_II_WITH_ARBORX:BOOL=ON' '-DARBORX_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/arborx-1.1-7feq6qw55a5vz7pbj7chszo6wednowk2' '-DDEAL_II_WITH_CGAL:BOOL=ON' '-DCGAL_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/cgal-5.4.1-bisxhj7znc4xqzaadnfuhgdpc4kayyal' '-DDEAL_II_WITH_ADOLC:BOOL=ON' '-DADOLC_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/adol-c-2.7.2-r6lsjzhut35x44zvc4swz66ug6othhu3' '-DDEAL_II_WITH_ARPACK:BOOL=ON' '-DARPACK_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/arpack-ng-3.8.0-2dx2qxyg7mj3bhg57eageyoaevb6edzv' '-DDEAL_II_ARPACK_WITH_PARPACK:BOOL=ON' '-DDEAL_II_WITH_NETCDF:BOOL=OFF' '-DDEAL_II_WITH_SCALAPACK:BOOL=ON' '-DSCALAPACK_FOUND:BOOL=ON' '-DSCALAPACK_INCLUDE_DIRS:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/netlib-scalapack-2.2.0-subtwjwg2ncyraoc7vesgzymzkllzdrd/include' '-DSCALAPACK_LIBRARIES:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/netlib-scalapack-2.2.0-subtwjwg2ncyraoc7vesgzymzkllzdrd/lib/libscalapack.so' '-DDEAL_II_WITH_OPENCASCADE:BOOL=ON' '-DOPENCASCADE_DIR:STRING=/home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/oce-0.18.3-65ain5ym3j33g4wtjrx7ftntd3fkin26' '-DDEAL_II_CXX_FLAGS:STRING=-march=skylake -mtune=skylake' '/tmp/amit/spack-stage/spack-stage-dealii-9.4.0-qyeizaptfk4artq6dukrk5scxsbdlbsc/spack-src'

1 error found in build log:
     630    --   HDF5_INCLUDE_DIRS: /home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/hdf5-1.10.7-li5cjehgcpyndpayoizmkefgixiflylp/include;/home/amit/Software/spack/opt/
            spack/linux-fedora36-skylake/gcc-12.1.1/openmpi-4.1.4-npqeruttdpenhdgkcqjm5lsvv6b5xacb/include
     631    --   HDF5_USER_INCLUDE_DIRS: /home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/hdf5-1.10.7-li5cjehgcpyndpayoizmkefgixiflylp/include;/home/amit/Software/spack
            /opt/spack/linux-fedora36-skylake/gcc-12.1.1/openmpi-4.1.4-npqeruttdpenhdgkcqjm5lsvv6b5xacb/include
     632    --   HDF5_LIBRARIES: /home/amit/Software/spack/opt/spack/linux-fedora36-skylake/gcc-12.1.1/hdf5-1.10.7-li5cjehgcpyndpayoizmkefgixiflylp/lib/libhdf5.so;/home/amit/Software/spack/
            opt/spack/linux-fedora36-skylake/gcc-12.1.1/openmpi-4.1.4-npqeruttdpenhdgkcqjm5lsvv6b5xacb/lib/libmpi.so
     633    -- Found HDF5
     634    -- Insufficient hdf5 installation found: hdf5 has to be configured with MPI support.
     635    -- DEAL_II_WITH_HDF5 has unmet external dependencies.
  >> 636    CMake Error at cmake/macros/macro_configure_feature.cmake:112 (MESSAGE):
     637    
     638    
     639      Could not find the hdf5 library!
     640    
     641      Insufficient hdf5 installation found!
     642    

See build log for details:
  /tmp/amit/spack-stage/spack-stage-dealii-9.4.0-qyeizaptfk4artq6dukrk5scxsbdlbsc/spack-build-out.txt
</pre></details>


### Information on your system

* **Spack:** 0.19.0.dev0 (b61187455afba7315e25f10a292c8554401e2156)
* **Python:** 3.10.6
* **Platform:** linux-fedora36-skylake
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/9298273/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/9298274/spack-build-out.txt)

Hi @jppelteret and @luca-heltai. I was trying to build `deal.ii` from the latest `spack` git version. The build spec shows that `hdf5` has been built with MPI on but still when configuring `deal.ii`, CMake complains that `hdf5` is missing MPI support.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/dealii/package.py
+++ b/var/spack/repos/builtin/packages/dealii/package.py
@@ -249,6 +249,19 @@ class Dealii(CMakePackage, CudaPackage):
         when="@9.2.0 ^boost@1.72.0:",
     )
 
+    # Fix issues due to override of CMake FIND_PACKAGE macro
+    # https://github.com/dealii/dealii/pull/14158/files
+    patch(
+        "https://github.com/dealii/dealii/commit/06bb9dc07efb6fea9912ee0d66264af548c552c8.patch?full_index=1",
+        sha256="8a1f7b9a155c8c496ce08b2abb1ba5d329b3b29169f36c11678aa4e3cebf97a2",
+        when="@9.4.0 ^hdf5",
+    )
+    patch(
+        "https://github.com/dealii/dealii/commit/40076ac1a013cd7d221f9dda913b4d0e6452c21e.patch?full_index=1",
+        sha256="7869dfab1116b6e862279bb6642c2c8fe49d87c42cfc6f031e03330f9f26a6c3",
+        when="@9.4.0 ^python",
+    )
+
     # Check for sufficiently modern versions
     conflicts("cxxstd=11", when="@9.3:")
 

